doc/images.md

```diff
@@ -13,29 +13,113 @@ flutter:
 
 It has to be a PNG file and it can have transparency.
 
+## Loading images
+
+Flame bundles an utility class called `Images` that allows you to easily load and cache images from the assets into the memory.
+
+Flutter has a collection of types related to images, and converting everything properly from a local asset to the Image that can be drawn on Canvas is a bit of a pain. This class allows you to obtain an Image that can be drawn on a Canvas using the `drawImageRect` method.
+
+It automatically caches any image loaded by filename, so you can safely call it many times.
+
+The methods for loading and clearing the cache are: `load`, `loadAll`, `clear` and `clearAll`. They return a `Future` for the Image loaded.
+
+To retrieve a previously cached image, the `fromCache` method can be used.
```

Comment:
1. To synchronously retrieve.
2. If an image with that key was not loaded prior, it will throw an exception.